contracts/external/cover/CoverBroker.sol

```diff
@@ -0,0 +1,194 @@
+// SPDX-License-Identifier: GPL-3.0-only
+
+pragma solidity ^0.8.18;
+
+import "@openzeppelin/contracts-v4/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts-v4/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts-v4/access/Ownable.sol";
+
+import "../../interfaces/ICover.sol";
+import "../../interfaces/ICoverBroker.sol";
+import "../../interfaces/IMemberRoles.sol";
+import "../../interfaces/IPool.sol";
+import "../../interfaces/INXMToken.sol";
+
+
+/// @title Cover Broker Contract
+/// @notice Enables non-members of the mutual to purchase cover policies.
+/// Supports ETH, NXM, and ERC20 asset payments which are supported by the pool.
+/// @dev See supported ERC20 payment methods via pool.getAssets.
+contract CoverBroker is ICoverBroker, Ownable {
+  using SafeERC20 for IERC20;
+
+  // Immutables
+  ICover public immutable cover;
+  IMemberRoles public immutable memberRoles;
+  IPool public immutable pool;
+  INXMToken public immutable nxmToken;
+  address public immutable tokenController;
+
+  // Constants
+  address public constant ETH = 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE;
+  uint private constant ETH_ASSET_ID = 0;
+  uint private constant NXM_ASSET_ID = type(uint8).max;
+
+  constructor(address _cover, address _memberRoles, address _pool, address _nxmToken, address _tokenController) {
+    cover = ICover(_cover);
+    memberRoles = IMemberRoles(_memberRoles);
+    pool = IPool(_pool);
+    nxmToken = INXMToken(_nxmToken);
+
+    tokenController = _tokenController;
+    nxmToken.approve(tokenController, type(uint256).max);
+  }
+
+  /// @notice Buys cover on behalf of the caller. Supports ETH, NXM and ERC20 asset payments which are supported by the pool.
+  /// @dev For NXM and ERC20 payments, ensure the Cover contract is approved to spend the tokens first (maxApproveCoverContract).
+  /// See supported ERC20 payment methods via pool.getAssets.
+  /// @param params The parameters required to buy cover.
+  /// @param poolAllocationRequests The allocation requests for the pool's liquidity.
+  /// @return coverId The ID of the purchased cover.
+  function buyCover(
+    BuyCoverParams calldata params,
```

Comment:
we should validate that params.owner is no the null address or the address of this contract (do the nft doesn't get stuck here)

Reply:
good catch. `params.owner` is not validated in the Cover contract